src/components/FilterControls/MediaTitleSearchInput/index.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react'
+import FormControl from '@mui/material/FormControl'
+import TextField from '@mui/material/TextField'
+import useDebounce from '../../../hooks/useDebounce'
+
+interface MediaTitleSearchInputProps {
+  defaultValue?: string
+  onChange: (searchTerm: string) => void
+}
+
+const MediaTitleSearchInput: React.FC<MediaTitleSearchInputProps> = ({
+  defaultValue,
+  onChange,
+}) => {
+  const [searchTerm, setSearchTerm] = React.useState<string | undefined>()
+  const debouncedSearchTerm = useDebounce(searchTerm, 400)
```

Comment:
I see what you're doing here and I do like it, but I think I can simplify things for you. I'd recommend debouncing the onChange call, _not_ debounced state updates.

The reason I say this is because you really don't need state here at all. If you stop passing a controlled `value` to your `TextField`, you can continue using the `onChange` handler, and the `TextField` will maintain its state all on its own.

For debouncing the actual `onChange` handling, I'd recommend something like lodash's `debounce` util, or rolling your own. This is also likely to make state management significantly easier to reason about.

Reply:
Check me on this, but the only issue I run into with this change is when the global Clear All FiIters button is clicked. I tried adding `useRef` so I could do `ref.current.value = ""` inside `handleClearFilters`, but the ref value is always undefined. 

Any ideas?


```
  const handleClearFilters = useCallback(() => {
    dispatch({ type: 'CLEAR_FILTERS' })

    searchInputRef.current.value = ''
  }, [])

```

```
const MediaTitleSearchInput: React.FC<MediaTitleSearchInputProps> = ({
  onChange,
  ref,
}) => {
  const handleChange = debounce((e: React.ChangeEvent<HTMLInputElement>) => {
    onChange(e.target.value)
  }, 400)

  return (
    <FormControl sx={{ m: 1, width: 300 }}>
      <TextField inputRef={ref} label="Title Search" onChange={handleChange} />
    </FormControl>
  )
}
```

Reply:
Oh I didn't think about clearing the search input. Good point!

You could either ignore me, _or_ hopefully fix this ref error via forwardRef: Hot take: I don't think you want to pass a default value here. In MUI at least, these values are usually reserved for something like a default numerical option in a field. Your fields don't have default options, and this is not something that ought to change over the lifetime of your app. I'd remove this prop. I'll repeat this elsewhere too.

Reply:
Oh I didn't think about clearing the search input, good point!

You could either ignore me, or try fixing the ref issue via forwardRef: https://react.dev/reference/react/forwardRef#forwardref